### PR TITLE
Text fields validated in all templates. 

### PR DESCRIPTION
**Pull request by free4murad**
+ Added strings to string resources file.
Fixed field validation issue in the following files and fields.

1. ComprehensionTemplate.java
a. Title
b. Passage
c. Timer
d. Question
e. Options 1,2,3 and 4
2. DictationTemplate.java
a. Title
b. Passage
3. FlashTemplate.java
a. Question
b. Answer
c. Answer hint
4. InfoTemplate.java
a. Word
b. Description
5. LearnSpellingTemplate.java
a. Word
b. Description
6. MatchTemplate.java
a. Title
b. First list title
c. Second list title
d. First list item
e. Second list item
7. QuizTemplate.java
a. Question
b. Option 1, 2, 3 and 4
8. VideoCollectionTemplate.java
a. Link
b. Title
c. Description
made the changes that were requested in #222

### CODE REVIEW

**Comment by free4murad** [comment]
@opticod 
please review this PR.
Also I have made the changes that were requested in PR#222

**Comment by free4murad** [comment]
@opticod 
Can you please explain both points.
If you could say what error you are getting I can solve faster.

**Comment by opticod** [comment]
1. For Match Template, you missed validation for meta's title. I can even create with empty meta title.
2. For Video Collection, by clicking OK in dialog box, the edit text area is filled by the contents of Text Hint.

**Comment by free4murad** [comment]
Solved the bug in MatchTemplate. It was because of function overloading.

Still not able to figure out the problem with VideoCollectionTemplate. Can you please explain it or upload a screen shot?

File: source-code/app/src/main/java/org/buildmlearn/toolkit/templates/VideoCollectionTemplate.java [hunk 1]
```diff
@@ -65,13 +66,15 @@ private static boolean validated(Context context, EditText link) {
             return false;
         }
 
-        String linkText = link.getText().toString();
+        String linkText = link.getText().toString().trim();
 
         if ("".equals(linkText)) {
-            Toast.makeText(context, R.string.video_collection_template_link_hint, Toast.LENGTH_SHORT).show();
+            link.setText(context.getString(R.string.video_collection_template_link_hint));
```
Comment (opticod) [inline source-code/app/src/main/java/org/buildmlearn/toolkit/templates/VideoCollectionTemplate.java]: @free4murad it will be link.setError()

**Comment by free4murad** [comment]
@opticod 
please check PR #253

**Pull request closed without merging**